vm/systemSmartContracts/delegation.go

```diff
@@ -827,7 +827,9 @@ func (d *delegation) delegate(args *vmcommon.ContractCallInput) vmcommon.ReturnC
 			return vmcommon.UserError
 		}
 
-		d.addDelegatorIfNew(dStatus, args.CallerAddr)
+		if isNew {
```

Comment:
make a function out of lines 823-837 addNewDelegatorToList which does the getDelegationStatus and setDelegationStatus and the append. The function should be called under the if isNew

Reply:
Done.